tests/sentry/api/endpoints/test_project_details.py

```diff
@@ -782,16 +844,74 @@ def test_invalid_digests_max_delay(self):
         assert self.project.get_option("digests:mail:minimum_delay") == min_delay
         assert self.project.get_option("digests:mail:maximum_delay") == max_delay
 
-    def test_dynamic_sampling_requires_feature_enabled(self):
-        self.get_error_response(
-            self.org_slug, self.proj_slug, dynamicSampling=_dyn_sampling_data(), status_code=403
-        )
+    def test_setting_dynamic_sampling_with_uniform_rule_and_basic_plan(self):
+        """
+        Tests that we can set a uniform sampling rule when basic dynamic sampling is enabled
+        """
+        config = _dyn_sampling_uniform_rule()
+
+        with Feature({"organizations:dynamic-sampling-basic": True}):
+            self.get_success_response(self.org_slug, self.proj_slug, dynamicSampling=config)
+
+        saved_config = self.project.get_option("sentry:dynamic_sampling")
+        assert config["rules"] == _ordered_dict_to_dict(saved_config)["rules"]
+
+    def test_setting_dynamic_sampling_with_uniform_rule_and_advanced_plan(self):
+        """
+        Tests that we can set a uniform sampling rule when advanced dynamic sampling is enabled
+        """
+        config = _dyn_sampling_uniform_rule()
+
+        with Feature(
+            {
+                "organizations:dynamic-sampling-basic": True,
+                "organizations:dynamic-sampling-advanced": True,
+            }
+        ):
+            self.get_success_response(self.org_slug, self.proj_slug, dynamicSampling=config)
+
+        saved_config = self.project.get_option("sentry:dynamic_sampling")
+        assert config["rules"] == _ordered_dict_to_dict(saved_config)["rules"]
+
+    def test_setting_dynamic_sampling_with_condition_rule_and_basic_plan(self):
+        """
+        Tests that we can set a condition sampling rule when basic dynamic sampling is enabled
```

Comment:
don't we want to 

```suggestion
        Tests that we can NOT set a condition sampling rule when ONLY basic dynamic sampling is enabled
```

?

Reply:
Oh yes, I made a mistake here! Thanks for the help

Reply:
np